app/web_framework/cluster_master_application.py

```diff
@@ -152,6 +156,51 @@ def get(self, build_id, subjob_id, atom_id):
         self.write(response)
 
 
+class _AtomConsoleHandler(_ClusterMasterBaseAPIHandler):
+    def get(self, build_id, subjob_id, atom_id):
+        """
+        :type build_id: int
+        :type subjob_id: int
+        :type atom_id: int
+        """
+        max_lines = int(self.get_query_argument('maxLines', 50))
+        offset_line = self.get_query_argument('offsetLine', None)
+
+        if offset_line is not None:
+            offset_line = int(offset_line)
+
+        try:
+            response = self._cluster_master.get_console_output(
+                build_id,
+                subjob_id,
+                atom_id,
+                Configuration['results_directory'],
+                max_lines,
+                offset_line
+            )
+            self.write(response)
+            return
+        except ItemNotFoundError as e:
+            # If the master doesn't have the atom's console output, it's possible it's currently being worked on,
+            # in which case the slave that is working on it may be able to provide the in-progress console output.
+            build = self._cluster_master.get_build(int(build_id))
+            subjob = build.subjob(int(subjob_id))
+            slave = subjob.slave
+
+            if slave is None:
+                raise e
+
+            api_url_builder = UrlBuilder(slave.url)
+            slave_console_url = api_url_builder.url(
+                'build/{}/subjob/{}/atom/{}/console'.format(build_id, subjob_id, atom_id))
+            slave_console_url += '?maxLines={}'.format(max_lines)
+
+            if offset_line is not None:
+                slave_console_url += '&offsetLine={}'.format(offset_line)
```

Comment:
Also, this is an annoying comment but I just noticed that our only other existing query parameters (for master and slave /eventlog) use underscores instead of camelCase.